src/main/scala/is/hail/utils/IntervalTree.scala

```diff
@@ -171,16 +218,16 @@ object IntervalTree {
     else {
       val mid = (start + end) / 2
       val (i, v) = intervals(mid)
-      val lft = fromSorted(pord, intervals, start, mid)
-      val rt = fromSorted(pord, intervals, mid + 1, end)
-      Some(IntervalTreeNode(i, lft, rt, {
-        val min1 = lft.map(x => pord.min(x.minimum, i.start)).getOrElse(i.start)
-        rt.map(x => pord.min(x.minimum, min1)).getOrElse(min1)
-      },
-        {
-          val max1 = lft.map(x => pord.max(x.maximum, i.end)).getOrElse(i.end)
-          rt.map(x => pord.max(x.maximum, max1)).getOrElse(max1)
-        }, v))
+      val left = fromSorted(pord, intervals, start, mid)
+      val right = fromSorted(pord, intervals, mid + 1, end)
+
+      val min = left.map { inode => inode.range }.getOrElse(i)
+      val eord = Interval.ordering(pord, startFirst=false)
+      val max = Array(left, right).flatten.foldLeft(i){ (i2, n) =>
```

Comment:
Prefer to foldLeft on left and then on right than to form array, flatten, and then foldLeft, to avoid copying left and right to make the array, and then copying them again to make the flattened array.